setup.py

```diff
@@ -8,20 +8,28 @@
 from setuptools import setup, find_packages
 
 
-def get_version():
+def load_version():
     """Returns the version found in nilearn/version.py
 
     Importing nilearn is not an option because there may dependencies
     like nibabel which are not installed and setup.py is supposed to
     install them.
     """
-    locals_dict = {}
+    # load all vars into globals, otherwise
+    #   the later function call using global vars doesn't work.
     globals_dict = {}
     with open(os.path.join('nilearn', 'version.py')) as fp:
-        exec(fp.read(), globals_dict, locals_dict)
+        exec(fp.read(), globals_dict)
 
-    return locals_dict['__version__']
+    return globals_dict
 
+
+def is_installing():
+    # Allow command-lines such as "python setup.py build install"
+    return len(sys.argv) > 1 and 'install' in sys.argv
```

Comment:
> I think the choice of using distutils vs setuptools can be made
> separately from this PR which is almost there in terms of fulfilling
> its original promise.

Yes! Agreed.

>   • I seem to remember distutils doesn't have find_packages, which makes it a
>     pain because you have to list all the subpackages by hand and it is quite
>     easy to forget to add a subpackage in setup.py next time each time we add a
>     subpackage.

That's a benefit benefit indeed. Setuptools do bring benefits.

>   • Full disclosure: setuptools was used in nilearn in 2012 but only for a
>     selection of commands that requires setuptools and 'install' is not one of
>     them

OK. Yes, this makes sens. That's the way it should be. In now way the
install mechanism of setuptools should be used. Setuptools should be used
for operations that cannot be done with distutils.

I'd really like to stress that while there are benefits, setuptools has
historically been very controversial, because it made choices that had
wide reaching impacts on the import mechanisms and the installation of
packages. There is a reason for which after 10 years it is not in the
standard library.

Installation is something that we really want robust and reliable. We do
not want to be making it brittle.